Frontend/ui-customer/src/Helpers/Utils.js

```diff
@@ -437,11 +437,21 @@ export const shuffle = function (array) {
 
 export const withinTimeRange = function (startTime) {
   return function (endTime) {
-    try {
-      var currentTimeString = moment(new Date()).format("HH:mm:ss");
-      return startTime < endTime ? between(currentTimeString, startTime, endTime) : between(currentTimeString, startTime, "23:59:59") || between(currentTimeString, "00:00:01", endTime);
-    }catch (err){
-      return false;
+    return function(timeStr){
+      try {
+        if(timeStr == "currentTime"){
+          var currentTimeString = moment(new Date()).format("HH:mm:ss");
+        return startTime < endTime ? between(currentTimeString, startTime, endTime) : between(currentTimeString, startTime, "23:59:59") || between(currentTimeString, "00:00:01", endTime);
+        }
+        else{
+          const time = moment(timeStr, 'h:mm A');
+          const startTime = moment('10:00 PM', 'h:mm A');
+          const endTime = moment('5:00 AM', 'h:mm A').add(1, 'day');
+          return time.isBetween(startTime, endTime, null, '[]');;
+        }
```

Comment:
@ashkriti can you optimize this func, here you can directly send timeStr as HH:mm:ss and no need for if-else conditions

Reply:
Done